static/app/views/dashboardsV2/utils.tsx

```diff
@@ -455,7 +455,7 @@ export function hasUnsavedFilterChanges(
     environment: new Set(currentFilters.environment),
   };
 
-  if (location.query?.release) {
+  if (defined(location.query?.release)) {
```

Comment:
Can we add this `defined` call to `filtersBar.tsx` where we pull the selected releases? It currently has this:
```js
  const selectedReleases =
    (location.query?.[DashboardFilterKeys.RELEASE]
      ? decodeList(location.query[DashboardFilterKeys.RELEASE])
      : filters?.[DashboardFilterKeys.RELEASE]) ?? [];
```
which will show `filters` in some cases when clearing the filters in a dirty state, or if you reload the URL with `release=`

Reply:
Steps for replicating the first case:
1. Have a dashboard with filters
2. Clear the filters, it'll clear as expected
3. Make some release selections, click out of the menu to apply them
4. Clear again, you'll see the original filters, but the URL has `release=` and you're in an unsaved filters state